Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -24,13 +24,81 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <AK/Function.h>
 #include <AK/StringBuilder.h>
 #include <LibJS/Heap/Heap.h>
 #include <LibJS/Runtime/GlobalObject.h>
 #include <LibJS/Runtime/PrimitiveString.h>
 #include <LibJS/Runtime/RegExpObject.h>
 #include <LibJS/Runtime/Value.h>
 
+static Flags options_from(const String& flags, JS::VM& vm, JS::GlobalObject& global_object)
```

Comment:
What's the reason for this not being in the `JS` namespace? It looks like it would belong there to me, and we could drop a whole bunch of `JS::` prefixes.

Reply:
Just something in my head saying "static helpers go in the global namespace or preferably an anonymous one in the global namespace"
I don't really mind putting it in the JS namespace, if you think that makes it better in any way, I'll move it there 🤷‍♂️

Reply:
I went ahead and put it in the JS namespace.

Reply:
ah, sorry I missed your comment! fwiw this was by no means an informed opinion, more of a gut feeling based on the (limited amount of) C++ code I have seen so far :smile: